### PR TITLE
[DDS-1121] Removed env conditional for ES config.

### PR DESCRIPTION
**Pull request by GROwen**
**edited to reflect updated PR for poc on content-vicpol**

### Related PRs
https://github.com/dpc-sdp/content-vicpol-vic-gov-au/pull/332
https://github.com/dpc-sdp/sdp-cmdb/pull/542
https://github.com/dpc-sdp/sdp-ansible-upgrade/pull/111
https://github.com/dpc-sdp/ansible-role-sdp-tide-platform/pull/19

 
 

# Checklist
- [x ] Affected Issues have been mentioned in the Closing issues section
- [x ] Documentation has been written/updated.
- [x ] Changelog entry has been written

Explain the **details** for making this change. What existing problem does the pull request solve?
Local and CI environments were excluded from values set for Elasticsearch in  `bay/settings.php` which meant the [change to remove elasticsearch_connector config](https://github.com/dpc-sdp/sdp-ansible-upgrade/pull/110/files) removed connectivity to the Elasticsearch container in those environments.

# Changelog Entry
Removes environment conditional for Elasticsearch configuration.

### CODE REVIEW

**Review by nicksantamaria — APPROVED** [review]

File: bay/images/docker/settings.php [hunk 1]
```diff
@@ -304,25 +304,23 @@
 $config['clamav.settings']['mode_daemon_tcpip']['port'] = $clamav_port;
 
 // Configure elasticsearch connections from environment variables.
-if (!in_array($lagoon_env_type, ['local', 'ci'])) {
-  if (getenv('SEARCH_HASH') && getenv('SEARCH_URL')) {
-    $config['elasticsearch_connector.cluster.elasticsearch_bay']['url'] = sprintf('http://%s.%s', getenv('SEARCH_HASH'), getenv('SEARCH_URL'));
-  }
+if (getenv('SEARCH_HASH') && getenv('SEARCH_URL')) {
+  $config['elasticsearch_connector.cluster.elasticsearch_bay']['url'] = sprintf('http://%s.%s', getenv('SEARCH_HASH'), getenv('SEARCH_URL'));
+}
```
Comment (nicksantamaria) [inline bay/images/docker/settings.php:311]:
```suggestion
} else {
  $config['elasticsearch_connector.cluster.elasticsearch_bay']['url'] =  "http://elasticsearch:9200";
}
```

**Review by nicksantamaria — APPROVED** [review]
LGTM!

**Comment by nicksantamaria** [comment]
Need to hold off merging this until `SEARCH_INDEX` has been added to production environment configmaps.

File: bay/images/docker/settings.php [hunk 2]
```diff
@@ -304,25 +304,32 @@
...
-  }
+if (getenv('SEARCH_INDEX')) {
+  $config['elasticsearch_connector.cluster.elasticsearch_bay']['options']['rewrite']['rewrite_index'] = 1;
+  $config['elasticsearch_connector.cluster.elasticsearch_bay']['options']['rewrite']['index'] = [
+    'prefix' => getenv('SEARCH_INDEX'),
+    'suffix' => '',
+  ];
+} else {
+  $config['elasticsearch_connector.cluster.elasticsearch_bay']['options']['rewrite']['index'] = [
+    'prefix' => 'elasticsearch_index_nonprod_',
```
Comment (nicksantamaria) [inline bay/images/docker/settings.php]: Lets change this value to something that works as a generic default value for nonprod and prod.

Perhaps `elasticsearch_index_default_` ?

**Review by nicksantamaria — CHANGES_REQUESTED** [review]

File: bay/images/docker/settings.php (hunk 2 above)
Comment (GROwen) [reply bay/images/docker/settings.php]: @nicksantamaria I've added the change to a more sane default value but before marking this as 'Needs Review' I'll  put up a corresponding CMDB PR to get this default set for projects.

Comment (GROwen) [reply bay/images/docker/settings.php]: Just clarifying this is doing away with the naming convention of `elasticsearch_index_pr` and replacing it with the suggested default.

Comment (GROwen) [reply bay/images/docker/settings.php]: @nicksantamaria I've updated CMDB values in this PR https://github.com/dpc-sdp/sdp-cmdb/pull/542, once that's deployed this should be safe to merge.

**Comment by GROwen** [comment]
@nicksantamaria Now that this is passing CI and https://github.com/dpc-sdp/sdp-cmdb/pull/542 is merged could you please review this again.

**Review by nicksantamaria — APPROVED** [review]

**Pull request merged**